Feature3

abhaumik13 opened the pull request:
Added Triage_Time_Analysis module, and the call in the run.py 
```
python run.py --feature 3
```
Fixes: #3 

Output:

Loaded 5573 issues from /Users/anirban/Desktop/UMD/Fall 2025/ENPM 611 - Software Engineering/Project Submissions/poetry_issues_all.json.
Triage time summary (days):
  count: 171
  mean_days: 231.3338436755469
  median_days: 19.852430555555554
  min_days: 0.0
  max_days: 1857.9858217592591
  std_days: 397.5317145630472

<img width="2940" height="1912" alt="image" src="https://github.com/user-attachments/assets/cd1ff465-a294-4a50-82c7-cd29578377b5" />

abhaumik13 committed:
Adding Triage Time Analysis

abhaumik13 committed:
Remove files not needed

aastha2912 commented:
Suggestions: 
1. It would be easier if this was pushed as a branch in the main repo instead of a fork. That way we can just checkout the branch and test it directly without setting up a whole separate repo.
2. Would be helpful if you could add screenshots or sample output in the PR itself - makes it easier to quickly see what the feature is doing.

aastha2912 commented:
Testing I did:

1. Ran it with the full issues JSON - works fine.
2. Also tried using --user flag - it runs, but it doesn’t actually filter anything. If this feature doesn't support --user, maybe the script should throw an error or at least print a message, so it’s not confusing.

Overall: Code works, just a couple of small suggestions. Good to merge after that.

orndorffgrant commented on run.py:
```diff
@@ -35,6 +36,8 @@ def parse_args():
                     help='Optional parameter for analyses focusing on a specific label')
     
+    ap.add_argument('--show-plot', action='store_true', help='Show plots when available')
+
```
I don't think we need this argument. It implies that all features will not show the plot unless this flag is used, but that is not true. Let's just default to showing the plot and remove this flag.

abhaumik13 replied:
Agreed with the observation. Made this change in code.

orndorffgrant commented on run.py:
```diff
@@ -52,6 +55,6 @@ def parse_args():
     pass # TODO call second analysis
 elif args.feature == 3:
-    pass # TODO call third analysis
+    TriageTimeAnalysis().triage_time_analysis(show_plot=args.show_plot)
```
To match the style of the others, let's make it look like this:
```suggestion
    TriageTimeAnalysis().run()
```

abhaumik13 replied:
Agreed with the observation. Made this change in code.

orndorffgrant commented on triage_time_analysis.py:
```diff
@@ -0,0 +1,94 @@
+        for e in sorted_events:
+            if not getattr(e, "event_type", None):
+                continue
+            et = str(e.event_type).strip().lower()
```
e.event_type will already be a lowercase stripped string
```suggestion
            et = e.event_type
```

abhaumik13 replied:
Updated code based on feedback

orndorffgrant commented on triage_time_analysis.py:
```diff
@@ -0,0 +1,94 @@
+                continue
+            et = str(e.event_type).strip().lower()
+            if et in ("assigned", "assign", "assignment", "status_change", "state_change"):
+                return e
```
None of these except "assigned" are real event types.
```suggestion
            if et == "assigned":
                return e
```

abhaumik13 replied:
Updated code based on feedback

orndorffgrant commented on triage_time_analysis.py:
```diff
@@ -0,0 +1,94 @@
+            et = str(e.event_type).strip().lower()
+            if et in ("assigned", "assign", "assignment", "status_change", "state_change"):
+                return e
+            if (getattr(e, "label", None) and "assign" in str(e.label).lower()) or \
+               (getattr(e, "comment", None) and "assign" in str(e.comment).lower()):
```
It is an interesting fallback that could lead to false positives. E.g. a comment "we should never triage or assign this" would be interpreted as a successful triage. I'll leave it up to you on whether you think this is worthwhile or not.

And technically, there is no need for `getattr` and also there are no labels with "assign" in them.
```suggestion
            if e.comment and "assign" in e.comment.lower():
```

abhaumik13 replied:
This makes sense. Have updated the code based on this feedback.

orndorffgrant reviewed (changes requested):
Looks great! Only blocking change is removing the `--show-plot` flag. The other comments are optional suggestions to remove unnecessary complexity.

abhaumik13 committed:
Update Triage Time Analysis Implementation

abhaumik13 commented:
I closed this PR and reopened a new PR on the same feature from a branch within the team repo instead of from a fork.
Also addressed all the comments in the new PR #7
